nixos/modules/services/networking/bind.nix

```diff
@@ -33,31 +50,176 @@ let
       ${cfg.extraConfig}
 
       ${ concatMapStrings
-          ({ name, file, master ? true, slaves ? [], masters ? [], extraConfig ? "" }:
+      (
+        { name, file, master, slaves, masters, extraConfig }:
+        # TODO: there has to be a better way:
+        # generate zone-file if file is an attrset, else use file
+        # checks if the attrs isn't a derivation incase something like pkgs.writeText was assigned to the file.
+          ''
+            zone "${name}" {
+              type ${if master then "master" else "slave"};
+              file "${if (builtins.isAttrs file) && !(isDerivation file) then zoneFile { inherit name; opts = file; } else file}";
```

Comment:
Is there a better way to solve this?

`file` was previously untyped, thus `str`, `path`, or `derivation` (which result in a path) was accepted (maybe even more types?)
Now `file` should allow the previous types OR the new submodule, what's the best way to check which one it is?